source/features/quick-reactions.tsx

```diff
@@ -0,0 +1,98 @@
+import delegate, {DelegateEvent} from 'delegate-it';
+import React from 'dom-chef';
+import * as pageDetect from 'github-url-detection';
+
+import features from '../feature-manager.js';
+
+const emojiSortOrder = ['THUMBS_UP', 'THUMBS_DOWN', 'LAUGH', 'HOORAY', 'CONFUSED', 'HEART', 'ROCKET', 'EYES'];
+
+function optimisticReactionFromPost(event: DelegateEvent<MouseEvent>): void {
+	const reaction = event.delegateTarget as HTMLButtonElement;
+	updateReaction(reaction);
+}
+
+function optimisticReactionFromMenu(event: DelegateEvent<MouseEvent>): void {
+	const reactionButton = event.delegateTarget as HTMLButtonElement;
+	const reactionsMenu = reactionButton.closest('reactions-menu')!;
+	const details = reactionsMenu.querySelector('details');
+	details!.open = false; // Close reactions menu immediately
+
+	const reactionsContainer = reactionsMenu.nextElementSibling!.querySelector('.js-comment-reactions-options')!;
```

Comment:
Positional traversing makes the code a little more brittle because a simple `span` in between will break it. If possible, select the parent element instead (e.g. `menu.parentElement.querySelector`) or just change what `reactionsMenu` to select its parent instead